ast/traversal.go

```diff
@@ -0,0 +1,14 @@
+package ast
+
+// searchCallback is the kind of function passed to DepthFirstSearch to get
+// notified about the traversed nodes in a gyp ast tree.
+type searchCallback func(expr Node, cbParam interface{})
+
+// DepthFirstSearch performs a depth-first traversal of expr's syntax tree and
+// invokes callback with the given cbParam for every node found.
+func DepthFirstSearch(expr Node, callback searchCallback, cbParam interface{}) {
```

Comment:
I'm not sure we actually need `cbParam`. We're only forwarding it to the callback function so the caller could use a function closure if additional vars are needed.